src/Week3/Golf.hs

```diff
@@ -0,0 +1,55 @@
+module Golf where
+
+import Data.List
+
+---------- ex 1 ----------
+
+skips :: [a] -> [[a]]
+skips l =  map (pickNth l) [1..(length l)]
+    where
+        pickNth :: [a] -> Int -> [a]
+        pickNth l n =
+            case (drop (n-1) l) of
+                x:xs -> x:(pickNth xs n)
+                []   -> []
+
+---------- ex 2 ----------
+
+localMaxima :: [Integer] -> [Integer]
+localMaxima (x:y:z:xs) =
+ if x<y && z<y
+ then y:(localMaxima (y:z:xs))
+ else (localMaxima (y:z:xs))
+localMaxima _ = []
+
+---------- ex 3 ----------
+
+histogram :: [Integer] -> String
+histogram l = hist l ++ "==========\n0123456789\n\n"
+    where
+        hist l@(x:xs)  = (hist $ foldl (flip id) l $ map delete [0..9]) ++ ((?) '*' ' ' <$> map (flip elem l) [0..9]) ++ "\n"
+        hist [] = ""
+
+
+
+
+---------- choice helper ----------
+(?) :: a -> a -> Bool -> a
+(?) t f b = if b then t else f
```

Comment:
And this is the same (almost) as `bool` function:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Bool.html#v:bool